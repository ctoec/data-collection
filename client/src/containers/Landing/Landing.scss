.Landing {
  .usa-hero {
    background-image: url('~@ctoec/component-library/dist/assets/images/hero.svg');
    color: white;
    padding: 3em 0;
    margin-bottom: 1em;

    h1 {
      font-weight: 400;
    }

    .btn--inverse {
      color: black;
      background-color: white;
      width: 10em;
      height: 3em;
      line-height: 150%;
    }
  }

  .footer {
    margin-top: -1em;
    margin-bottom: -10em;
    fill: white;

    .usa-button:hover {
      background-color: #2378c3;
    }
  }
}
