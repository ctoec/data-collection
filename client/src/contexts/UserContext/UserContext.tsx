import React, {
  Dispatch,
  SetStateAction,
  useContext,
  useEffect,
  useState,
} from 'react';
import AuthenticationContext from '../AuthenticationContext/AuthenticationContext';
import { User } from '../../shared/models';
import { apiGet, apiPost } from '../../utils/api';
import { Moment as MomentInterface } from 'moment';
import * as moment from 'moment';

export type UserContextType = {
  user: User | null;
  loading: boolean;
  setConfidentialityAgreedDate: Dispatch<SetStateAction<Date>>;
  confidentialityAgreedDate: MomentInterface | null;
};

const UserContext = React.createContext<UserContextType>({
  user: null,
  loading: true,
  confidentialityAgreedDate: null,
  setConfidentialityAgreedDate: () => {},
});

const { Provider, Consumer } = UserContext;

export type UserProviderPropsType = {};

/**
 * Context Provider for injecting User object into React hierarchy
 *
 * @param props Props with user
 */
const UserProvider: React.FC<UserProviderPropsType> = ({ children }) => {
  const { accessToken, loading } = useContext(AuthenticationContext);
  const [user, setUser] = useState<User | null>(null);
  const [userLoading, setUserLoading] = useState(loading);
  const [refetchUser, setRefetchUser] = useState(false);

  useEffect(() => {
    if (accessToken) {
      setUserLoading(true);
      apiGet('users/current', accessToken)
        .then((data) => {
          setUser(data);
        })
        .catch((err) => {
          // This doesn't need to throw an error because the lack of user is taken to mean that this failed
          console.error(err);
        })
        .finally(() => {
          setUserLoading(false);
          setRefetchUser(false);
        });
    } else {
      setUserLoading(loading);
    }
  }, [accessToken, loading, refetchUser]);

  function setConfidentialityAgreedDate() {
    apiPost(
      `users/current`,
      { ...user, confidentialityAgreedDate: moment.utc() },
      { accessToken, jsonParse: false }
    )
      .then(() => setRefetchUser(true))
      .catch((err) => {
        console.error(err);
        throw new Error('Error processing confidentiality agreement');
      });
  }

  return (
    <Provider
      value={{
        loading: userLoading,
        user,
        confidentialityAgreedDate: user?.confidentialityAgreedDate || null,
        setConfidentialityAgreedDate,
      }}
    >
      {children}
    </Provider>
  );
};

export { UserProvider };
export { Consumer as UserConsumer };
export default UserContext;
