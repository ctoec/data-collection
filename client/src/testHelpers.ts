import React from 'react';
import { RenderResult, render, cleanup, act } from '@testing-library/react';
import { axe } from 'jest-axe';

export const accessibilityTestHelper = (
  component: React.ReactElement,
  before?: (_: RenderResult) => Promise<void>
) => {
  it('passes AXE accessibility checks', async () => {
    await act(async () => {
      const renderResult = render(component);
      if (before) {
        await before(renderResult);
      }
      const results = await axe(renderResult.container);
      expect(results).toHaveNoViolations();
    });
    cleanup();
  });
};

export const snapshotTestHelper = (component: React.ReactElement) => {
  it('matches snapshot', async () => {
    await act(async () => {
      const { asFragment } = render(component);
      expect(asFragment()).toMatchSnapshot();
    });
  });
};
