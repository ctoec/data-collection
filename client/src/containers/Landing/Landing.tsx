import React, { useContext } from 'react';
import { Redirect } from 'react-router-dom';
import { ReactComponent as ArrowRight } from '@ctoec/component-library/dist/assets/images/arrowForward.svg';
import {
  Button,
  LoadingWrapper,
  Reviews,
  TextWithIcon,
} from '@ctoec/component-library';
import HomeCareerBubbleSrc from '@ctoec/component-library/dist/assets/images/homeCareerBubble.png';
import UserContext from '../../contexts/UserContext/UserContext';
import { getH1RefForTitle } from '../../utils/getH1RefForTitle';

const Landing: React.FC = () => {
  const { user, loading } = useContext(UserContext);
  const h1Ref = getH1RefForTitle();

  // To prevent flash of splash page if user is logged in
  if (loading) {
    return <LoadingWrapper loading={true} />;
  }

  // If the user is logged in, don't show the splash page
  if (user) {
    return user.isAdmin ? (
      <Redirect to="/overview/" />
    ) : (
      <Redirect to="/roster/" />
    );
  }

  return (
    <div className="Landing">
      <div className="usa-hero">
        <div className="grid-container">
          <h1 ref={h1Ref} className="margin-bottom-2">
            Welcome to ECE Reporter
          </h1>
          <Button
            className="btn--inverse"
            href="/login"
            text="Log in"
            appearance="base"
          />
        </div>
      </div>
      <div className="grid-container margin-top-4">
        <div className="grid-row grid-gap">
          <div className="tablet:grid-col-3">
            <img className="hero-bubble" src={HomeCareerBubbleSrc} alt="" />
          </div>
          <div className="tablet:grid-col-6">
            <h2 className="text-primary text-light margin-y-3">
              We support affordable child care in Connecticut
            </h2>
            <p className="line-height-sans-5">
              Publicly-funded early care and education programs use ECE Reporter
              to share data with the Connecticut Office of Early Childhood
              (OEC).
            </p>
            <p>
              The Office of Early Childhood uses this data to pay programs and
              help families access quality care.
            </p>
            <div className="margin-bottom-4">
              <h2 className="text-primary text-light margin-y-3">Learn more</h2>
              <h3>See the data collected by ECE Reporter</h3>
              <Button
                appearance="unstyled"
                href="/data-requirements"
                className="margin-bottom-3 display-block"
                text={
                  <TextWithIcon
                    text="Learn the data requirements"
                    Icon={ArrowRight}
                    direction="right"
                    iconSide="right"
                    className="text-underline"
                  />
                }
              />
              <Button
                appearance="unstyled"
                href="/template"
                className="margin-bottom-3 display-block"
                text={
                  <TextWithIcon
                    text="See the latest data template"
                    Icon={ArrowRight}
                    direction="right"
                    iconSide="right"
                    className="text-underline"
                  />
                }
              />
              <h3>Get help with ECE Reporter</h3>
              <Button
                appearance="unstyled"
                external={true}
                href="https://help.ece-reporter.ctoec.org"
                className="margin-bottom-3 display-block"
                text={
                  <TextWithIcon
                    text="View how-to guides"
                    Icon={ArrowRight}
                    direction="right"
                    iconSide="right"
                    className="text-underline"
                  />
                }
              />
              <h3>Learn more about OEC</h3>
              <Button
                appearance="unstyled"
                external={true}
                href="https://ctoec.org"
                className="margin-bottom-3 display-block"
                text={
                  <TextWithIcon
                    text="Visit OEC's website"
                    Icon={ArrowRight}
                    direction="right"
                    iconSide="right"
                    className="text-underline"
                  />
                }
              />
              <h3>Learn how ECE Reporter collects and stores data</h3>
              <Button
                appearance="unstyled"
                href="/privacy"
                className="margin-bottom-3 display-block"
                text={
                  <TextWithIcon
                    text="See the privacy policy"
                    Icon={ArrowRight}
                    direction="right"
                    iconSide="right"
                    className="text-underline"
                  />
                }
              />
            </div>
          </div>
        </div>
      </div>
      <footer className="bg-base-lightest padding-y-6 footer">
        <div className="grid-container">
          <div className="grid-row grid-gap">
            <div className="grid-col-9">
              <h2 className="text-primary text-light margin-y-3">
                Make ECE Reporter better for providers like you
              </h2>
              <p>
                Take a quick survey to let the team know how this tool can work
                better for you.
              </p>
              <a
                className="usa-button usa-button--outline margin-top-3"
                href="https://forms.monday.com/forms/f5bcf5d3d611b99f45ffb8187cd6c2b9?r=use1"
                target="_blank"
              >
                Give Feedback
              </a>
            </div>
            <div className="grid-col-3 margin-y-3" role="presentation">
              <Reviews />
            </div>
          </div>
        </div>
      </footer>
    </div>
  );
};

export default Landing;
