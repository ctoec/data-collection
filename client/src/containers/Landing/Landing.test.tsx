import React from 'react';
import Landing from './Landing';
import { snapshotTestHelper, accessibilityTestHelper } from '../../testHelpers';
import UserContext from '../../contexts/UserContext/UserContext';
import moment from 'moment';

const LandingWithUserProvider = (
  <UserContext.Provider
    value={{
      loading: false,
      user: null,
      confidentialityAgreedDate: moment.utc(),
      setConfidentialityAgreedDate: () => {},
    }}
  >
    <Landing />
  </UserContext.Provider>
);

describe('Landing', () => {
  snapshotTestHelper(LandingWithUserProvider, { wrapInRouter: true });
  accessibilityTestHelper(LandingWithUserProvider, { wrapInRouter: true });
});
