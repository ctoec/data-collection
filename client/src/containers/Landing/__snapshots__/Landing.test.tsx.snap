// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Landing"
  >
    <div
      class="usa-hero"
    >
      <div
        class="grid-container"
      >
        <h1
          class="margin-bottom-2"
        >
          Welcome to ECE Reporter
        </h1>
        <a
          class="usa-button usa-button--base btn--inverse"
          href="/login"
        >
          Log in
        </a>
      </div>
    </div>
    <div
      class="grid-container margin-top-4"
    >
      <div
        class="grid-row grid-gap"
      >
        <div
          class="tablet:grid-col-3"
        >
          <img
            alt=""
            class="hero-bubble"
            src="homeCareerBubble.png"
          />
        </div>
        <div
          class="tablet:grid-col-6"
        >
          <h2
            class="text-primary text-light margin-y-3"
          >
            We support affordable child care in Connecticut
          </h2>
          <p
            class="line-height-sans-5"
          >
            Publicly-funded early care and education programs use ECE Reporter to share data with the Connecticut Office of Early Childhood (OEC).
          </p>
          <p>
            The Office of Early Childhood uses this data to pay programs and help families access quality care.
          </p>
          <div
            class="margin-bottom-4"
          >
            <h2
              class="text-primary text-light margin-y-3"
            >
              Learn more
            </h2>
            <h3>
              See the data collected by ECE Reporter
            </h3>
            <a
              class="usa-button usa-button--unstyled margin-bottom-3 display-block"
              href="/data-requirements"
            >
              <span
                class="oec-text-with-icon text-underline"
              >
                Learn the data requirements
                <svg
                  class="oec-text-with-icon__icon oec-text-with-icon__icon--direction-right oec-text-with-icon__icon--side-right"
                >
                  arrowForward.svg
                </svg>
              </span>
            </a>
            <a
              class="usa-button usa-button--unstyled margin-bottom-3 display-block"
              href="/template"
            >
              <span
                class="oec-text-with-icon text-underline"
              >
                See the latest data template
                <svg
                  class="oec-text-with-icon__icon oec-text-with-icon__icon--direction-right oec-text-with-icon__icon--side-right"
                >
                  arrowForward.svg
                </svg>
              </span>
            </a>
            <h3>
              Learn more about OEC
            </h3>
            <a
              class="usa-button usa-button--unstyled margin-bottom-3 display-block"
              href="https://ctoec.org"
            >
              <span
                class="oec-text-with-icon text-underline"
              >
                Visit OEC's website
                <svg
                  class="oec-text-with-icon__icon oec-text-with-icon__icon--direction-right oec-text-with-icon__icon--side-right"
                >
                  arrowForward.svg
                </svg>
              </span>
            </a>
            <h3>
              Learn how ECE Reporter collects and stores data
            </h3>
            <a
              class="usa-button usa-button--unstyled margin-bottom-3 display-block"
              href="/privacy"
            >
              <span
                class="oec-text-with-icon text-underline"
              >
                See the privacy policy
                <svg
                  class="oec-text-with-icon__icon oec-text-with-icon__icon--direction-right oec-text-with-icon__icon--side-right"
                >
                  arrowForward.svg
                </svg>
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="bg-base-lightest padding-y-6 footer"
    >
      <div
        class="grid-container"
      >
        <div
          class="grid-row grid-gap"
        >
          <div
            class="grid-col-9"
          >
            <h2
              class="text-primary text-light margin-y-3"
            >
              Make ECE Reporter better for providers like you
            </h2>
            <p>
              Take a quick survey to let the team know how this tool can work better for you.
            </p>
            <a
              class="usa-button usa-button--outline"
              href="https://forms.monday.com/forms/f5bcf5d3d611b99f45ffb8187cd6c2b9?r=use1"
              target="_blank"
            >
              Give Feedback
            </a>
          </div>
          <div
            class="grid-col-3 margin-y-3"
            role="presentation"
          >
            <svg
              height="142"
              viewBox="0 0 212 142"
              width="212"
            >
              <g
                fill="none"
                fill-rule="nonzero"
              >
                <path
                  d="M174.01 84.6a2.92 2.92 0 0 1 .44-.09l5.926-11.701-1.81-2.636L182.303 67 187 72.585l-9.567 13.529a2.703 2.703 0 0 1-.59 3.084 2.902 2.902 0 0 1-3.179.554c-1.075-.475-1.735-1.547-1.658-2.692.077-1.145.875-2.124 2.005-2.46z"
                  fill="#A0616A"
                />
                <path
                  d="M183.68 69.181a1.177 1.177 0 0 1-.263.213l-5.825 3.445a1.15 1.15 0 0 1-.935.107c-.312-.1-.57-.327-.71-.629l-2.833-6.107a1.202 1.202 0 0 1 .4-1.497l3.61-2.506a1.153 1.153 0 0 1 1.487.144l5.048 5.167c.446.458.456 1.194.022 1.663z"
                  fill="#CCC"
                />
                <path
                  d="M192 136.21l-3.168.79L184 126.167l4.675-1.167z"
                  fill="#A0616A"
                />
                <path
                  d="M186.595 136.93l6.03-1.641 1.034 3.798-9.828 2.675a3.937 3.937 0 0 1 2.764-4.832z"
                  fill="#2F2E41"
                />
                <path
                  d="M165 138h-2.71L161 126h4z"
                  fill="#A0616A"
                />
                <path
                  d="M159.864 139H166v3h-10c0-.796.407-1.559 1.132-2.121.724-.563 1.707-.879 2.732-.879z"
                  fill="#2F2E41"
                />
                <ellipse
                  cx="168.5"
                  cy="51"
                  fill="#A0616A"
                  rx="6.5"
                  ry="6"
                />
                <path
                  d="M166.205 130H161.8a1.19 1.19 0 0 1-1.182-1.117c-1.313-20.995-.545-36.055 2.416-47.395.013-.05.03-.099.048-.146l.236-.594c.179-.453.614-.75 1.099-.748h9.06c.345 0 .673.152.898.416l1.23 1.444c.024.028.047.057.068.087 5.55 7.785 10.522 23.004 16.28 43.46a1.188 1.188 0 0 1-.777 1.46l-4.735 1.524a1.187 1.187 0 0 1-1.426-.617l-12.286-25.637a.92.92 0 0 0-1.741.277l-3.61 26.557a1.19 1.19 0 0 1-1.173 1.029z"
                  fill="#2F2E41"
                />
                <path
                  d="M174.679 81h-.038l-11.18-.335c-.594-.015-1.086-.44-1.156-1-1.417-10.858 2.47-13.767 4.206-14.514a.88.88 0 0 0 .538-.694l.15-1.138c.026-.2.108-.392.238-.552 2.723-3.34 6.441-4.409 8.416-4.75.323-.054.656.019.922.203.265.184.439.463.48.771l.226 1.635a.876.876 0 0 0 .19.427c5.68 7.028-.648 17.405-1.974 19.414-.22.331-.604.533-1.018.533z"
                  fill="#CCC"
                />
                <path
                  d="M181.655 51.757a25.708 25.708 0 0 0-4.956-5.49c-1.575-1.301-3.329-2.413-5.263-2.945-1.934-.532-4.07-.442-5.835.55-1.765.993-3.59 2.964-3.601 5.079-.003.679.16 1.598.792 1.48l.183.026 4.576-.978 1.949 7.155c.137.503.32 1.065.771 1.28.425.201.916.011 1.346-.179.07-.03.678-1.473.678-1.473a.258.258 0 0 1 .198-.175.247.247 0 0 1 .24.097l.36.477a.49.49 0 0 0 .593.16l7.463-3.295c.343-.152.728-.35.827-.73.096-.363-.114-.731-.32-1.04z"
                  fill="#2F2E41"
                />
                <path
                  d="M211.736 142h-79.472c-.146 0-.264-.224-.264-.5s.118-.5.264-.5h79.472c.146 0 .264.224.264.5s-.118.5-.264.5z"
                  fill="#3F3D56"
                />
                <path
                  d="M168.967 101a.793.793 0 0 1-.48-.165l-1.156-.875a.543.543 0 0 0-.662 0l-1.156.875a.791.791 0 0 1-.96 0 .869.869 0 0 1-.297-.95l.442-1.414a.597.597 0 0 0-.205-.654l-1.156-.875a.869.869 0 0 1-.297-.95.82.82 0 0 1 .777-.586h1.43c.243 0 .46-.163.535-.405l.441-1.414A.82.82 0 0 1 167 93a.82.82 0 0 1 .777.587l.442 1.414a.564.564 0 0 0 .534.405h1.43a.82.82 0 0 1 .777.587c.11.35-.01.733-.297.95l-1.156.873a.597.597 0 0 0-.205.655l.442 1.414a.855.855 0 0 1-.297.95.793.793 0 0 1-.48.165z"
                  fill="#FFF"
                />
                <path
                  d="M172.551 97.484a2.758 2.758 0 0 0 1.21-2.035 2.725 2.725 0 0 0-.835-2.204L175 69l-6.174.822-.118 23.556a2.724 2.724 0 0 0 .038 3.706c.98 1.047 2.615 1.219 3.805.4z"
                  fill="#A0616A"
                />
                <path
                  d="M175.634 72a1.22 1.22 0 0 1-.333-.046l-6.461-1.824a1.159 1.159 0 0 1-.72-.581 1.082 1.082 0 0 1-.041-.902l2.391-5.929c.206-.517.776-.811 1.34-.691l4.268.89c.55.115.937.59.922 1.129l-.199 6.861c-.018.607-.534 1.091-1.167 1.093z"
                  fill="#CCC"
                />
                <path
                  d="M114.15 103H1.85A1.845 1.845 0 0 1 0 101.163V67.837C0 66.823.829 66 1.85 66h112.3a1.845 1.845 0 0 1 1.85 1.837v33.326a1.845 1.845 0 0 1-1.85 1.837zM1.85 66.525c-.73 0-1.322.588-1.322 1.312v33.326c0 .724.592 1.311 1.322 1.312h112.3c.73 0 1.322-.588 1.322-1.312V67.837c0-.724-.592-1.311-1.322-1.312H1.85z"
                  fill="#3F3D56"
                />
                <path
                  d="M51.46 117a.992.992 0 0 1-.6-.207l-1.447-1.092a.679.679 0 0 0-.826 0l-1.446 1.092a.99.99 0 0 1-1.2 0 1.086 1.086 0 0 1-.371-1.187l.552-1.768a.746.746 0 0 0-.255-.817l-1.446-1.093a1.086 1.086 0 0 1-.371-1.187c.137-.438.529-.734.97-.734h1.788a.705.705 0 0 0 .669-.505l.552-1.768c.137-.438.529-.734.971-.734.442 0 .834.296.97.734l.553 1.768a.705.705 0 0 0 .669.505h1.787c.442 0 .834.296.971.734.137.437-.013.916-.37 1.187l-1.447 1.093a.746.746 0 0 0-.255.817l.552 1.768a1.07 1.07 0 0 1-.37 1.187.992.992 0 0 1-.6.207zM65.46 117a.992.992 0 0 1-.6-.207l-1.447-1.092a.679.679 0 0 0-.826 0l-1.446 1.092a.99.99 0 0 1-1.2 0 1.086 1.086 0 0 1-.371-1.187l.552-1.768a.746.746 0 0 0-.255-.817l-1.446-1.093a1.086 1.086 0 0 1-.371-1.187c.137-.438.529-.734.97-.734h1.788a.705.705 0 0 0 .669-.505l.552-1.768c.137-.438.529-.734.971-.734.442 0 .834.296.97.734l.553 1.768a.705.705 0 0 0 .669.505h1.787c.442 0 .834.296.971.734.137.437-.013.916-.37 1.187l-1.447 1.093a.746.746 0 0 0-.255.817l.552 1.768a1.07 1.07 0 0 1-.37 1.187.992.992 0 0 1-.6.207zM81.46 117a.992.992 0 0 1-.6-.207l-1.447-1.092a.68.68 0 0 0-.826 0l-1.446 1.092a.989.989 0 0 1-1.2 0 1.086 1.086 0 0 1-.371-1.187l.552-1.768a.746.746 0 0 0-.255-.817l-1.446-1.093a1.086 1.086 0 0 1-.371-1.187c.137-.438.529-.734.97-.734h1.788a.705.705 0 0 0 .669-.505l.552-1.768c.137-.438.529-.734.971-.734.442 0 .834.296.97.734l.553 1.768a.705.705 0 0 0 .669.505h1.787c.442 0 .834.296.971.734.137.437-.013.916-.37 1.187l-1.447 1.093a.746.746 0 0 0-.255.817l.552 1.768a1.07 1.07 0 0 1-.37 1.187.992.992 0 0 1-.6.207zM95.46 117a.992.992 0 0 1-.6-.207l-1.447-1.092a.679.679 0 0 0-.826 0l-1.446 1.092a.99.99 0 0 1-1.2 0 1.086 1.086 0 0 1-.371-1.187l.552-1.768a.746.746 0 0 0-.255-.817l-1.446-1.093a1.086 1.086 0 0 1-.371-1.187c.137-.438.529-.734.97-.734h1.788a.705.705 0 0 0 .669-.505l.552-1.768c.137-.438.529-.734.971-.734.442 0 .834.296.97.734l.553 1.768a.705.705 0 0 0 .669.505h1.787c.442 0 .834.296.971.734.137.437-.013.916-.37 1.187l-1.447 1.093a.746.746 0 0 0-.255.817l.552 1.768a1.07 1.07 0 0 1-.37 1.187.992.992 0 0 1-.6.207z"
                  fill="#61366F"
                />
                <path
                  d="M106.59 117a1.07 1.07 0 0 1-.647-.222 1.15 1.15 0 0 1-.4-1.277l.542-1.732a.626.626 0 0 0-.214-.687l-1.418-1.07a1.168 1.168 0 0 1-.4-1.277c.148-.471.57-.79 1.046-.79h1.752a.592.592 0 0 0 .562-.424l.542-1.732c.147-.47.569-.789 1.045-.789s.898.319 1.045.79l.542 1.731a.592.592 0 0 0 .562.424h1.752c.476 0 .898.32 1.045.79.147.47-.014.986-.4 1.277l-1.417 1.07a.626.626 0 0 0-.214.687l.541 1.732c.147.47-.014.986-.4 1.277-.384.29-.906.29-1.291 0l-1.418-1.07a.571.571 0 0 0-.694 0l-1.418 1.07a1.068 1.068 0 0 1-.646.222zm2.41-1.939c.232 0 .458.077.646.219l1.418 1.07a.573.573 0 0 0 .694 0 .628.628 0 0 0 .215-.686l-.542-1.732c-.148-.47.014-.987.4-1.277l1.417-1.07a.628.628 0 0 0 .215-.687.593.593 0 0 0-.562-.424h-1.752c-.477 0-.9-.318-1.046-.79l-.541-1.731a.593.593 0 0 0-.562-.425.593.593 0 0 0-.562.425l-.541 1.731c-.147.472-.57.79-1.046.79H105.1a.593.593 0 0 0-.562.424c-.079.253.008.53.215.686l1.418 1.07c.385.291.547.807.399 1.278l-.542 1.732c-.079.253.008.53.215.686a.573.573 0 0 0 .694 0l1.418-1.07a1.07 1.07 0 0 1 .646-.219z"
                  fill="#3F3D56"
                />
                <path
                  d="M15.71 74c-.944 0-1.71.672-1.71 1.5s.766 1.5 1.71 1.5h86.58c.944 0 1.71-.672 1.71-1.5s-.766-1.5-1.71-1.5H15.71zM15.71 83c-.944 0-1.71.672-1.71 1.5s.766 1.5 1.71 1.5h86.58c.944 0 1.71-.672 1.71-1.5s-.766-1.5-1.71-1.5H15.71zM15.71 92c-.944 0-1.71.672-1.71 1.5s.766 1.5 1.71 1.5h86.58c.944 0 1.71-.672 1.71-1.5s-.766-1.5-1.71-1.5H15.71z"
                  fill="#CCC"
                />
                <path
                  d="M15.71 74c-.944 0-1.71.672-1.71 1.5s.766 1.5 1.71 1.5h86.58c.944 0 1.71-.672 1.71-1.5s-.766-1.5-1.71-1.5H15.71zM15.71 83c-.944 0-1.71.672-1.71 1.5s.766 1.5 1.71 1.5h86.58c.944 0 1.71-.672 1.71-1.5s-.766-1.5-1.71-1.5H15.71zM15.71 92c-.944 0-1.71.672-1.71 1.5s.766 1.5 1.71 1.5h86.58c.944 0 1.71-.672 1.71-1.5s-.766-1.5-1.71-1.5H15.71z"
                  fill="#CCC"
                />
                <path
                  d="M114.15 37H1.85A1.845 1.845 0 0 1 0 35.163V1.837C0 .823.829 0 1.85 0h112.3A1.845 1.845 0 0 1 116 1.837v33.326A1.845 1.845 0 0 1 114.15 37zM1.85.525c-.73 0-1.322.588-1.322 1.312v33.326c0 .724.592 1.311 1.322 1.312h112.3c.73 0 1.322-.588 1.322-1.312V1.837c0-.724-.592-1.311-1.322-1.312H1.85z"
                  fill="#3F3D56"
                />
                <path
                  d="M51.46 51a.992.992 0 0 1-.6-.207l-1.447-1.092a.679.679 0 0 0-.826 0l-1.446 1.092a.99.99 0 0 1-1.2 0 1.086 1.086 0 0 1-.371-1.187l.552-1.768a.746.746 0 0 0-.255-.817l-1.446-1.093a1.086 1.086 0 0 1-.371-1.187c.137-.438.529-.734.97-.734h1.788a.705.705 0 0 0 .669-.505l.552-1.768c.137-.438.529-.734.971-.734.442 0 .834.296.97.734l.553 1.768a.705.705 0 0 0 .669.505h1.787c.442 0 .834.296.971.734.137.437-.013.916-.37 1.187l-1.447 1.093a.746.746 0 0 0-.255.817l.552 1.768a1.07 1.07 0 0 1-.37 1.187.992.992 0 0 1-.6.207zM65.46 51a.992.992 0 0 1-.6-.207l-1.447-1.092a.679.679 0 0 0-.826 0l-1.446 1.092a.99.99 0 0 1-1.2 0 1.086 1.086 0 0 1-.371-1.187l.552-1.768a.746.746 0 0 0-.255-.817l-1.446-1.093a1.086 1.086 0 0 1-.371-1.187c.137-.438.529-.734.97-.734h1.788a.705.705 0 0 0 .669-.505l.552-1.768c.137-.438.529-.734.971-.734.442 0 .834.296.97.734l.553 1.768a.705.705 0 0 0 .669.505h1.787c.442 0 .834.296.971.734.137.437-.013.916-.37 1.187l-1.447 1.093a.746.746 0 0 0-.255.817l.552 1.768a1.07 1.07 0 0 1-.37 1.187.992.992 0 0 1-.6.207zM81.46 51a.992.992 0 0 1-.6-.207l-1.447-1.092a.68.68 0 0 0-.826 0l-1.446 1.092a.989.989 0 0 1-1.2 0 1.086 1.086 0 0 1-.371-1.187l.552-1.768a.746.746 0 0 0-.255-.817l-1.446-1.093a1.086 1.086 0 0 1-.371-1.187c.137-.438.529-.734.97-.734h1.788a.705.705 0 0 0 .669-.505l.552-1.768c.137-.438.529-.734.971-.734.442 0 .834.296.97.734l.553 1.768a.705.705 0 0 0 .669.505h1.787c.442 0 .834.296.971.734.137.437-.013.916-.37 1.187l-1.447 1.093a.746.746 0 0 0-.255.817l.552 1.768a1.07 1.07 0 0 1-.37 1.187.992.992 0 0 1-.6.207z"
                  fill="#61366F"
                />
                <path
                  d="M106.59 51a1.07 1.07 0 0 1-.647-.222 1.15 1.15 0 0 1-.4-1.277l.542-1.732a.626.626 0 0 0-.214-.687l-1.418-1.07a1.168 1.168 0 0 1-.4-1.277c.148-.471.57-.79 1.046-.79h1.752a.592.592 0 0 0 .562-.424l.542-1.732c.147-.47.569-.789 1.045-.789s.898.319 1.045.79l.542 1.731a.592.592 0 0 0 .562.424h1.752c.476 0 .898.32 1.045.79.147.47-.014.986-.4 1.277l-1.417 1.07a.626.626 0 0 0-.214.687l.541 1.732c.147.47-.014.986-.4 1.277-.384.29-.906.29-1.291 0l-1.418-1.07a.571.571 0 0 0-.694 0l-1.418 1.07a1.068 1.068 0 0 1-.646.222zm2.41-1.939c.232 0 .458.077.646.219l1.418 1.07a.573.573 0 0 0 .694 0 .628.628 0 0 0 .215-.686l-.542-1.732c-.148-.47.014-.987.4-1.277l1.417-1.07a.628.628 0 0 0 .215-.687.593.593 0 0 0-.562-.424h-1.752c-.477 0-.9-.318-1.046-.79l-.541-1.731a.593.593 0 0 0-.562-.425.593.593 0 0 0-.562.425l-.541 1.731c-.147.472-.57.79-1.046.79H105.1a.593.593 0 0 0-.562.424c-.079.253.008.53.215.686l1.418 1.07c.385.291.547.807.399 1.278l-.542 1.732c-.079.253.008.53.215.686a.573.573 0 0 0 .694 0l1.418-1.07a1.07 1.07 0 0 1 .646-.219zM90.59 51a1.07 1.07 0 0 1-.647-.222 1.15 1.15 0 0 1-.4-1.277l.542-1.732a.626.626 0 0 0-.214-.687l-1.418-1.07a1.168 1.168 0 0 1-.4-1.277c.148-.471.57-.79 1.046-.79h1.752a.592.592 0 0 0 .562-.424l.542-1.732c.147-.47.569-.789 1.045-.789s.898.319 1.045.79l.542 1.731a.592.592 0 0 0 .562.424H96.9c.476 0 .898.32 1.045.79.147.47-.014.986-.4 1.277l-1.417 1.07a.626.626 0 0 0-.214.687l.541 1.732c.147.47-.014.986-.4 1.277-.384.29-.906.29-1.291 0l-1.418-1.07a.571.571 0 0 0-.694 0l-1.418 1.07a1.068 1.068 0 0 1-.646.222zM93 49.061c.232 0 .458.077.646.219l1.418 1.07a.573.573 0 0 0 .694 0 .628.628 0 0 0 .215-.686l-.542-1.732c-.148-.47.014-.987.4-1.277l1.417-1.07a.628.628 0 0 0 .215-.687.593.593 0 0 0-.562-.424h-1.752c-.477 0-.9-.318-1.046-.79l-.541-1.731a.593.593 0 0 0-.562-.425.593.593 0 0 0-.562.425l-.541 1.731c-.147.472-.57.79-1.046.79H89.1a.593.593 0 0 0-.562.424c-.079.253.008.53.215.686l1.418 1.07c.386.291.547.807.399 1.278l-.542 1.732c-.079.253.008.53.215.686a.573.573 0 0 0 .694 0l1.418-1.07a1.07 1.07 0 0 1 .646-.219z"
                  fill="#3F3D56"
                />
                <path
                  d="M15.71 8C14.767 8 14 8.672 14 9.5s.766 1.5 1.71 1.5h86.58c.944 0 1.71-.672 1.71-1.5s-.766-1.5-1.71-1.5H15.71zM15.71 17c-.944 0-1.71.672-1.71 1.5s.766 1.5 1.71 1.5h86.58c.944 0 1.71-.672 1.71-1.5s-.766-1.5-1.71-1.5H15.71zM15.71 26c-.944 0-1.71.672-1.71 1.5s.766 1.5 1.71 1.5h86.58c.944 0 1.71-.672 1.71-1.5s-.766-1.5-1.71-1.5H15.71z"
                  fill="#CCC"
                />
                <path
                  d="M15.71 8C14.767 8 14 8.672 14 9.5s.766 1.5 1.71 1.5h86.58c.944 0 1.71-.672 1.71-1.5s-.766-1.5-1.71-1.5H15.71zM15.71 17c-.944 0-1.71.672-1.71 1.5s.766 1.5 1.71 1.5h86.58c.944 0 1.71-.672 1.71-1.5s-.766-1.5-1.71-1.5H15.71zM15.71 26c-.944 0-1.71.672-1.71 1.5s.766 1.5 1.71 1.5h86.58c.944 0 1.71-.672 1.71-1.5s-.766-1.5-1.71-1.5H15.71z"
                  fill="#CCC"
                />
              </g>
            </svg>
          </div>
        </div>
        <div />
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
